runtime/compiler/optimizer/VectorAPIExpansion.cpp

```diff
@@ -1128,13 +1750,82 @@ TR_VectorAPIExpansion::expandVectorAPI()
          TR::VectorLength vectorLength = OMR::DataType::bitsToVectorLength(bitsLength);
 
          TR_ASSERT_FATAL(methodTable[handlerIndex]._methodHandler(this, treeTop, node, elementType, vectorLength, numLanes, checkMode),
-                         "Analysis should've proved that method is supported");
+                         "Analysis should've proved that method %p is supported for %s", node, (checkMode == checkScalarization) ? "scalarization" : "vectorization");
 
-         TR::Node::recreate(parent, TR::treetop);
-         methodTable[handlerIndex]._methodHandler(this, treeTop, node, elementType, vectorLength, numLanes, doMode);
+         if (!checkBoxing)
+            {
+            _nodeTable[node->getGlobalIndex()]._origSymRef = node->getSymbolReference();
+            TR::Node::recreate(parent, TR::treetop);
+            }
+
+         if (boxingAllowed())
+            {
+            // Unbox operands if needed, before calling handler
+            int32_t numChildren = node->getNumChildren();
+
+            if (_trace)
+               traceMsg(comp(), "Checking if children of vectorized node %p need to be unboxed\n", node);
+
+            for (int32_t i = 0; i < numChildren; i++)
+               {
+               // TO DO: check Mask type through the method table
+               if ((i >= getFirstOperandIndex(methodSymbol) &&
+                   i < (getFirstOperandIndex(methodSymbol) + getNumOperands(methodSymbol))) ||
+                   (i == getMaskIndex(methodSymbol) && node->getChild(i)->getOpCodeValue() != TR::aconst))
+                  {
+                  TR::Node *operand = node->getChild(i);
+                  bool vectorized = false;
+
+                  {
+                  TR::DataType elementType;
```

Comment:
Please use a different variable name at least for `elementType` because another `elementType` is declared in the outer scope.

Reply:
Done